src/processTargets/modifiers/scopeHandlers/TokenScopeHandler.ts

```diff
@@ -23,4 +24,17 @@ export default class TokenScopeHandler extends NestedScopeHandler {
         }),
     }));
   }
+
+  isPreferredOver(
+    scopeA: TargetScope,
+    _scopeB: TargetScope,
+  ): boolean | undefined {
+    const {
+      editor: { document },
+    } = scopeA;
+    const { identifierMatcher } = getMatcher(document.languageId);
+
+    const textA = document.getText(scopeA.domain);
+    return identifierMatcher.test(textA) ? true : undefined;
```

Comment:
I'm still not sure what's the different between false and undefined is?

Reply:
Undefined lets the containing scope stage decide; didn't want to bake the fact that we go right into this function.  Tho on second thought, your optimisations already rely heavily on that fact.  How about now?

Reply:
Looks good